plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/EditMavenArtifactDialog.java

```
@@ -19,103 +19,204 @@
+                UIConnectionMessages.dialog_edit_driver_edit_maven_ignore_transient_dependencies,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies_tip,
+                originalArtifact != null && originalArtifact.isIgnoreDependencies(),
+                2);
+            ignoreDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    ignoreDependencies = ignoreDependenciesCheckbox.getSelection();
+                }
+            });
 
+            Button loadOptionalDependenciesCheckbox = UIUtils.createCheckbox(settingsGroup,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies,
+                UIConnectionMessages.dialog_edit_driver_edit_maven_load_optional_dependencies_tip,
+                originalArtifact != null && originalArtifact.isIgnoreDependencies(),
+                2);
+            loadOptionalDependenciesCheckbox.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    loadOptionalDependencies = loadOptionalDependenciesCheckbox.getSelection();
+                }
+            });
+        }
+
+        return composite;
+    }
+
+    private void parseArtifactText() {
+        try {
+            artifacts.clear();
+            if (REGEX_FOR_GRADLE.matcher(fieldText.getText()).find()) {
+                artifacts.addAll(parseGradle());
+            } else {
+                artifacts.addAll(parseMaven());
+            }
+            setStatus(false, NLS.bind(UIConnectionMessages.dialog_edit_driver_edit_maven_artifacts_count, artifacts.size()));
+        } catch (Exception e) {
+            setStatus(true, e.getMessage());
+            log.trace("Error parsing artifact!", e);
```

This effectively eats the exception since `trace` is disabled by default. I think we need to include this exception in debug log.